### PR TITLE
fix errors related to string labels

### PR DESCRIPTION
**Pull request by JKL98ISR**
fixed issue in Model Error Analysis and Calibration Score

### CODE REVIEW

File: deepchecks/checks/performance/model_error_analysis.py [hunk 1]
```diff
@@ -147,11 +148,13 @@ def run(self, train_dataset: Dataset, test_dataset: Dataset, model=None) -> Chec
...
-            def scoring_func(dataset):
+            def scoring_func(dataset: Dataset):
                 return per_sample_mse(dataset.label_col, model.predict(dataset.features_columns))
         else:
-            def scoring_func(dataset):
-                return per_sample_binary_cross_entropy(dataset.label_col,
+            def scoring_func(dataset: Dataset):
+                le = preprocessing.LabelEncoder()
+                encoded_label = le.fit_transform(dataset.label_col)
+                return per_sample_binary_cross_entropy(encoded_label,
```
Comment (nirhutnik) [inline deepchecks/checks/performance/model_error_analysis.py:157]: Why not do the encoding outside of the function? I think that the implicit assumption that LabelEncoder.fit_transform will give the same result for train and test might cause bugs in some future.

File: deepchecks/checks/performance/calibration_score.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def _calibration_score(self, dataset: Dataset, model):
         briers_scores = {}
 
         if len(dataset.classes) == 2:
-            briers_scores[0] = brier_score_loss(ds_y, y_pred[:, 1])
+            briers_scores[0] = brier_score_loss(ds_y, y_pred[:, 1], pos_label=dataset.classes[1])
```
Comment (nirhutnik) [inline deepchecks/checks/performance/calibration_score.py:55]: Still needed if the results are passed as [0,1]?

**Review by nirhutnik — CHANGES_REQUESTED** [review]

File: deepchecks/checks/performance/model_error_analysis.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/checks/performance/model_error_analysis.py:158]: did a fix

File: deepchecks/checks/performance/calibration_score.py (hunk 2 above)
Comment (JKL98ISR) [reply deepchecks/checks/performance/calibration_score.py:55]: no but it doesn't matter

**Review by nirhutnik — APPROVED** [review]
